bugfix/KAS-3221 Pub v3: Reference document link ontbreekt

mdv-professional opened the pull request:
['t kasticketje](https://kanselarij.atlassian.net/browse/KAS-3221)

---

#### Binnen scope
- Optie 1: verholpen
- Optie 2: niet verholpen
  Het is al eerder ter sprake gekomen dat er een probleem is in de applicatie bij voortijdig afbreken van een batch-save. Bij mijn weten wordt deze mogelijkheid nergens tegengegaan bij errors bij het saven (vb. rollback).
- Optie 3: verholpen

#### Buiten scope
##### Gedaan
- null-pointer-exception bij afwezigheid van document-type.
- PowerSelect `onopen` en `oninput` argumenten zijn hernoemd naar nieuwe API `onOpen` en `onInput`: ik heb de callbacks verwijderd: toen ze terug aangeroepen werden, bleken ze niet meer te werken zoals ik ze bedoeld had in de `PublicationFlowSelector`.

Commit:
aside/bugfix allow documentType to be undefined
bugfix/KAS-3221 pub: v3: reference document link

Commit:
do not allow save to be interrupted by closing modal
bugfix/KAS-3221 pub: v3: reference document link

Commit:
Merge branch 'development' into bugfix/KAS-3221-pub--v3--reference-document-link

Commit:
:lint-fix
bugfix/KAS-3221 pub: v3: reference document link

Commit:
do not allow save to be interrupted by closing modal (preventing cancel)
bugfix/KAS-3221 pub: v3: reference document link

ghost commented:
Ik krijg dit precies op nog een manier gereproduceerd. Ik check straks. (nu lunch)

Commit:
(lint-fix)
bugfix/KAS-3221 pub: v3: reference document link

Commit:
prevent possible concurrency issue in linking/unlinking of reference documents
bugfix/KAS-3221 pub: v3: reference document link

Commit:
(aside/bugfix) new callback casing for PowerSelect
bugfix/KAS-3221 pub: v3: reference document link

Commit:
(aside/bugfix) make search behavior consistent
note: related to the previous commit
reason: The `onOpen` and `onInput` and `search` actions and the `options` property didn't work well togehter anymore because of changes to the `@search` attribute callback. For simplicity I fell back to a simple EmberPowerSelect configuration with `@search`.
The drawback is it does not show a list of related publication-flows when opening the selector. It did function more or less this way, because of the new casing of EmberPowerSelect's callback arguments, which had "disabled" the event actions (see previous commit).
bugfix/KAS-3221 pub: v3: reference document link

Commit:
(lint-fix)
bugfix/KAS-3221 pub: v3: reference document link

ValenberghsSven reviewed (approved):
Dit werkt, krijg het niet meteen stuk.
Ik heb nog niet multi-user/concurrency getest, maar dat moeten we algemeen eens doen door die publicatie views en niet hier.

Iets dat ik zou kunnen aangeven buiten scope is dat de zoekfunctie in die batch-documents-publication-modal redelijk beperkt is.
Je kan enkel bestaande publicaties zien die reeds gekoppeld zijn aan het dossier.
Dus in de praktijk is het misschien wel handig dat je meteen al enkele resultaten ziet en niet hoeft te zoeken omdat er meestal niet zoveel zijn. Dat lijkt nu wel weg te vallen

Zoeken zou dan enkel nut hebben als je een dossier hebt met 100den decreten/besluiten en je meerdere aan 1 publicatie wilt koppelen.

ghost commented:
@ValenberghsSven @erikap Over de zoekfunctie
Dit lijkt me ook logischer. Ik heb erovernagedacht hoe ik het zou aanpakken, maar leek me te erg buiten scope. Ik weet niet juist waar ik het neerschrijf, mocht dit ooit aangepakt worden. Best lijkt me het `@options`-argument mee te geven aan de `PowerSelect`. Dit is de lijst items die wordt weergegeven wanneer de `PowerSelect` geen tekst invoer heeft. Deze wordt volgens mij best gequeried in de bovenliggende modal-component om te vermijden dat de query wordt uitgevoerd per document.
